flixkit.go

```diff
@@ -7,63 +7,27 @@ import (
 	"io"
 	"log"
 	"net/http"
-	"regexp"
+	"os"
 )
 
-type Network struct {
-	Address        string `json:"address"`
-	FqAddress      string `json:"fq_address"`
-	Contract       string `json:"contract"`
-	Pin            string `json:"pin"`
-	PinBlockHeight uint64 `json:"pin_block_height"`
-}
-
-type Argument struct {
-	Index    int      `json:"index"`
-	Type     string   `json:"type"`
-	Messages Messages `json:"messages"`
-	Balance  string   `json:"balance"`
-}
-
-type Title struct {
-	I18N map[string]string `json:"i18n"`
-}
-
-type Description struct {
-	I18N map[string]string `json:"i18n"`
-}
-
-type Messages struct {
-	Title       *Title       `json:"title,omitempty"`
-	Description *Description `json:"description,omitempty"`
-}
-
-type Dependencies map[string]Contracts
-type Contracts map[string]Networks
-type Networks map[string]Network
-type Arguments map[string]Argument
-
-type Data struct {
-	Type         string       `json:"type"`
-	Interface    string       `json:"interface"`
-	Messages     Messages     `json:"messages"`
-	Cadence      string       `json:"cadence"`
-	Dependencies Dependencies `json:"dependencies"`
-	Arguments    Arguments    `json:"arguments"`
-}
-
-type FlowInteractionTemplate struct {
-	FType    string `json:"f_type"`
-	FVersion string `json:"f_version"`
-	ID       string `json:"id"`
-	Data     Data   `json:"data"`
-}
 
 type FlixService interface {
 	GetFlixRaw(ctx context.Context, templateName string) (string, error)
 	GetFlix(ctx context.Context, templateName string) (*FlowInteractionTemplate, error)
 	GetFlixByIDRaw(ctx context.Context, templateID string) (string, error)
 	GetFlixByID(ctx context.Context, templateID string) (*FlowInteractionTemplate, error)
+	GenFlixBinding(ctx context.Context, templateID string, lang string, isLocal bool) (string, error)
+}
+
+// OsFileReader is a real implementation that calls os.ReadFile.
+type OsFileReader struct{}
```

Comment:
ths fs abstraction https://pkg.go.dev/io/fs kind of has this built in already there is even a MapFs that can be used in tests.